packages/transaction-controller/src/TransactionController.ts

```diff
@@ -767,17 +767,23 @@ export class TransactionController extends BaseControllerV1<
       validateGasValues(gasValues);
     }
 
-    log('Creating cancel transaction', transactionId, gasValues);
-
+    await this.pendingTransactionTracker.checkTransactions();
     const transactionMeta = this.getTransaction(transactionId);
     if (!transactionMeta) {
       return;
     }
 
+    // Transaction already confirmed, do not create a cancel transaction.
+    if (transactionMeta.status === TransactionStatus.confirmed) {
```

Comment:
> From the controller perspective, we asked it to do something and it can't do it so it feels conceptually like an error.

I agree, but I have concerns if we only pass the error to the client and leave it to handle. (Assuming we don't want back and forth and prevent this happening `stopTransaction` => `error passed to client` => `client asks if tx is confirmed` => ... )

Also in case of an error, client could show some error stating that "this transaction is confirmed, you cannot cancel it". But if we don't force update of this transaction status, client will keep showing it as "submitted". 

In my opinion the best here is that we can send some error to the client(so it can show "too late" message) but also update, does it sound good on your end?